EasyPost/Services/Beta/SmartRateService.cs

```diff
@@ -0,0 +1,106 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EasyPost._base;
+using EasyPost.Http;
+using EasyPost.Models.API;
+using EasyPost.Parameters.SmartRate;
+using EasyPost.Utilities.Internal.Attributes;
+
+namespace EasyPost.Services.Beta
+{
+    /// <summary>
+    ///     Class representing a set of SmartRate-related beta functionality.
+    /// </summary>
+    // ReSharper disable once ClassNeverInstantiated.Global
+    public class SmartRateService : EasyPostService
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="SmartRateService" /> class.
+        /// </summary>
+        /// <param name="client">The <see cref="EasyPostClient"/> to tie to this service and use for API calls.</param>
+        internal SmartRateService(EasyPostClient client)
+            : base(client)
+        {
+        }
+
+        #region CRUD Operations
+
+        /// <summary>
+        ///     Get the <see cref="SmartRate"/>s for a <see cref="Shipment"/>.
+        ///     <a href="https://www.easypost.com/docs/api#retrieve-time-in-transit-statistics-across-all-rates-for-a-shipment">Related API documentation</a>.
+        /// </summary>
+        /// <param name="shipmentId">The ID of the <see cref="Shipment"/> to get rates for.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/> to use for the HTTP request.</param>
+        /// <returns>A list of <see cref="SmartRate"/>s.</returns>
+        [CrudOperations.Read]
+        public async Task<List<SmartRate>> GetSmartRates(string shipmentId, CancellationToken cancellationToken = default)
+        {
+            return await RequestAsync<List<SmartRate>>(Method.Get, $"shipments/{shipmentId}/smartrate", cancellationToken, rootElement: "result");
+        }
+
+        /// <summary>
+        ///     Retrieve the estimated delivery date of each rate for a <see cref="Shipment"/> via the SmartRates API, based on a specific ship date.
+        ///     <a href="https://www.easypost.com/docs/api#retrieve-estimated-delivery-date-and-total-transit-days-across-all-rates-for-a-shipment">Related API documentation</a>.
```

Comment:
URL seems incorrect to what this function does, you might want to add a TODO and come back to fix it when API docs is added for this endpoint